### PR TITLE
[tobias-loew-flags] add new port

### PR DESCRIPTION
**Pull request by jiayuehua**

### CODE REVIEW

**Comment by LilyWangLL** [comment]
Usage passed on x64-windows.

**Review by LilyWangLL — DISMISSED** [review]

File: ports/bext-flags/add-install-configuration.patch [hunk 1]
```diff
@@ -0,0 +1,75 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 78d96ff..9cf5fbe 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.12...3.30)
+ project(boost_flags)
+ list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+ 
++include(CMakePackageConfigHelpers)
```
Comment (dg0yt) [inline ports/bext-flags/add-install-configuration.patch:9]: Put this where it is used. Smaller patch.

File: ports/bext-flags/add-install-configuration.patch [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ # C++ standard version selection
+ ##################################################
+@@ -63,9 +64,42 @@ endif()
+ ##################################################
+ # flags sub-library
+ ##################################################
+-add_library(boost_flags INTERFACE)
+-#target_include_directories(boost_flags PUBLIC ${Boost_INCLUDE_DIRS}) 
+-target_include_directories(boost_flags INTERFACE ${CMAKE_SOURCE_DIR}/include)
++add_library(flags INTERFACE)
```
Comment (dg0yt) [inline ports/bext-flags/add-install-configuration.patch:20]: It is not necessary to rename the lib.
You can set the EXPORT_NAME property if you want to change it.
(I acknowledge that it is a INTERFACE lib, not a binary.)

File: ports/bext-flags/add-install-configuration.patch [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
++  cmake/unofficial-bext-flags-config.cmake.in
++  "${CMAKE_CURRENT_BINARY_DIR}/cmake/unofficial-bext-flags-config.cmake"
++  INSTALL_DESTINATION lib/cmake/unofficial-bext-flags/
++  NO_CHECK_REQUIRED_COMPONENTS_MACRO)
++
++
++# Export.
++export(
++  TARGETS flags
++  FILE "${CMAKE_CURRENT_BINARY_DIR}/cmake/unofficial-bext-flags-targets.cmake")
```
Comment (dg0yt) [inline ports/bext-flags/add-install-configuration.patch:37]: Not needed for vcpkg.

File: ports/bext-flags/add-install-configuration.patch [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
++
++# Install.
++install(TARGETS flags EXPORT unofficial-bext-flags)
++
++install(
++  EXPORT unofficial-bext-flags
++  NAMESPACE unofficial-bext::
```
Comment (dg0yt) [inline ports/bext-flags/add-install-configuration.patch:44]: Canonical namespace:
```suggestion
+  NAMESPACE unofficial::bext-flags::
```
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#add-cmake-exports-in-an-unofficial--namespace

File: ports/bext-flags/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-${PORT} CONFIG_PATH "lib/cmake/unofficial-${PORT}")
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib")
```
Comment (dg0yt) [inline ports/bext-flags/portfile.cmake:20]: Install the CMake config to `share/unofficial-bext-flags`, and you can remove `CONFIG_PATH` and `file(REMOVE_RECURSe ...)`.

**Comment by JavierMatosD** [comment]
Please apply @dg0yt's suggestions. Marking this PR draft for now. Mark "ready for review" when ready.

File: ports/bext-flags/portfile.cmake (hunk 5 above)
Comment (jiayuehua) [reply ports/bext-flags/portfile.cmake]: just remove those lines.

File: ports/bext-flags/add-install-configuration.patch (hunk 4 above)
Comment (jiayuehua) [reply ports/bext-flags/add-install-configuration.patch]: just rename it.

File: ports/bext-flags/add-install-configuration.patch (hunk 3 above)
Comment (jiayuehua) [reply ports/bext-flags/add-install-configuration.patch]: just delete those  lines.

File: ports/bext-flags/add-install-configuration.patch (hunk 2 above)
Comment (jiayuehua) [reply ports/bext-flags/add-install-configuration.patch]: fine.

**Comment by JavierMatosD** [comment]
I'm tagging team review on account that this is a boost extension.

**Comment by JavierMatosD** [comment]
@BillyONeal, @ras0219-msft, and I discussed this. 

We note that the current `bext-` ports in the registry are associated with either https://github.com/boost-ext/ or https://github.com/qlibs. 

To avoid confusion, please drop the `bext-` prefix and replace with some variation of upstream org name. (e.g. `tobias-flags`) @tobias-loew, any preference?

@tzlaine, I noticed we added a `bext-text` port recently, but it doesn't appear to be associates with the `boost-ext` org or the `qlibs` org? If it is not, we should rename the port to reflect your organization.

@laudrup, is the port `best-wintls` associated with either `boost-ext` or `qlibs`? If not, we should rename the port so that the prefix reflects your organization.

**Comment by laudrup** [comment]
@JavierMatosD 

I'm not responsible for the `best-wintls` package and wasn't even aware that it existed.

I am of course happy that someone took the effort to create a package of my library and will be happy to help supporting it.

Regarding your question I'm not related to any organization. I don't what would be the best prefix for this package and I don't have any experience with VCPKG packaging.

The wintls library is an extension to Asio which is part of Boost but can also be used standalone. Not sure if that helps with categorizing it.

**Comment by tobias-loew** [comment]
Similar to @laudrup I'm not affiliated an organization. Please choose a prefix that works best.

The flags-library is not yet part of Boost. I'm planning to get it there, but it will take a while.

**Comment by jiayuehua** [comment]
@tobias-loew then I rename the port name to tobias-loew-flags?

**Comment by BillyONeal** [comment]
Rerunning due to network mistake last night.

**Pull request merged**